tests/test_theming/test_theming.py

```diff
@@ -126,3 +148,41 @@ def test_theme_sidebars(app, status, warning):
     assert '<h3>Related Topics</h3>' not in result
     assert '<h3>This Page</h3>' not in result
     assert '<h3 id="searchlabel">Quick search</h3>' in result
+
+
+@pytest.mark.parametrize(
+    'theme_name',
+    [
+        'alabaster',
+        'agogo',
+        'basic',
+        'bizstyle',
+        'classic',
+        'default',
+        'epub',
+        'haiku',
+        'nature',
+        'nonav',
+        'pyramid',
+        'scrolls',
+        'sphinxdoc',
+        'traditional',
+    ],
+)
+def test_theme_builds(make_app, rootdir, sphinx_test_tempdir, theme_name):
+    """Test all the themes included with Sphinx build a simple project and produce valid XHTML."""
+    testroot_path = rootdir / 'test-basic'
+    srcdir = sphinx_test_tempdir / f'test-theme-{theme_name}'
+    shutil.copytree(testroot_path, srcdir)
+
+    app = make_app(srcdir=srcdir, confoverrides={'html_theme': theme_name})
+    app.build()
+    assert not app.warning.getvalue().strip()
+    assert app.outdir.joinpath('index.html').exists()
+
+    # check that the generated HTML files are well-formed (as strict XHTML)
```

Comment:
Worth being a little careful about this statement I think - we're confirming that it's valid `XML`, but not valid `XHTML` (...but maybe we could do later?).

Reply:
changed `XHTML` to `XML` 👍